zapcore/sampler.go

```diff
@@ -81,17 +81,87 @@ func (c *counter) IncCheckReset(t time.Time, tick time.Duration) uint64 {
 	return 1
 }
 
+// SamplingDecision represents a decision made by sampler.
+type SamplingDecision uint8
+
+const (
+	// LogDropped means that a log was dropped.
+	LogDropped SamplingDecision = iota
+	// LogSampled means that a log was successfully sampled.
+	LogSampled
+)
+
+// optionFunc wraps a func so it satisfies the SamplerOption interface.
+type optionFunc func(*sampler)
+
+func (f optionFunc) apply(s *sampler) {
+	f(s)
+}
+
+// SamplerOption configures a Sampler option.
+type SamplerOption interface {
+	apply(*sampler)
+}
+
+// NopSamplingHook is the default hook used by sampler.
+func NopSamplingHook(_ Entry, _ SamplingDecision) {}
+
+// SamplerHook registers a function  which will be called when Sampler makes a
+// decision. Currently a hook is called when a log is dropped and
+// zapcore.LogDropped decision is emitted.
```

Comment:
Second part no longer true.
```suggestion
// decision.
```